src/uu/nl/src/nl.rs

```diff
@@ -106,6 +106,42 @@ impl NumberFormat {
     }
 }
 
+enum SectionDelimiter {
+    Header,
+    Body,
+    Footer,
+    None,
+}
+
+struct SectionDelimiterParser {
+    delimiter: String,
+}
+
+impl SectionDelimiterParser {
+    fn new(delimiter: &str) -> Self {
+        let delimiter = if delimiter.chars().count() == 1 {
```

Comment:
```suggestion
        let delimiter = if delimiter.len() == 1 {
```

Reply:
Hm, `len` returns the number of bytes, not the number of chars (for example, `"ä".len()` returns `2`). And my intention was to get the number of chars. On the other hand, using `len` leads to the same output as GNU nl when specifying `ä` as delimiter. Which is incorrect, in my opinion ;-) I guess I have to ask in the GNU coreutils project what the expected output should be in such a case.

Reply:
Ohh I see. Maybe it should even be unicode width?